### PR TITLE
[ParticleMechanics] Particle generator fix

### PR DESCRIPTION
**Pull request by bodhinandach**
I decided to change to the default shape function value as the function MP16ShapeFunctions() returns me the size of (16,3) which leads to bad index when I do a loop from a tetrahedral mesh with 4 nodes. The test is modified accordingly and GiDInterface will be modified too.

### CODE REVIEW

**Comment by philbucher** [comment]
fine for me, however I would like to leave approval to @antonialarese

**Comment by bodhinandach** [comment]
@antonialarese as discussed. I also changed it to switch case to make it a bit cleaner.

File: applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h [hunk 1]
```diff
@@ -217,40 +217,62 @@ class MPMStrategy
...
+                                if (TDim==2) {
+                                    shape_functions_values = this->MP33ShapeFunctions();
+                                    break;
+                                }
+                            default:
+                                std::string warning_msg = "The input number of particle: " + std::to_string(m_NumPar);
+                                warning_msg += " is not available for Triangular" + std::to_string(TDim) + "D.\n";
+                                warning_msg += "Available options are: 1, 3, 6, 12, 16 (only 2D), and 33 (only 2D).\n";
+                                warning_msg += "The default number of particle: 3 is currently assumed.";
+                                KRATOS_INFO("MPM_Strategy") << "WARNING: " << warning_msg << std::endl;
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h:249]: if it is a warning you should use KRATOS_WARNING

Comment (antonialarese) [reply applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h:249]: @bodhinandach  and @philbucher: as far as I understand you reached an agreement on leaving KRATOS_INFO for the readability of the output file. 
If that is the case, the commit is fine for me. We already discussed it with Nanda.

**Review by antonialarese — APPROVED** [review]

**Pull request merged**